agent.py

```diff
@@ -56,10 +59,64 @@ def run_multimodal_agent(ctx: JobContext, participant: rtc.RemoteParticipant):
         model=model,
         chat_ctx=chat_ctx,
     )
+
     agent.start(ctx.room, participant)
 
-    # to enable the agent to speak first
-    agent.generate_reply()
+    if participant.attributes.get("lk.agent.pre-connect-audio"):
+        logger.info("registering pre-connect audio handler")
+        ctx.room.register_byte_stream_handler("lk.agent.pre-connect-audio-buffer", lambda reader, participant_identity: asyncio.create_task(handle_pre_connect(agent, model.sessions[0], reader)))
+    else:
+        logger.info("unregistering pre-connect audio handler")
+        ctx.room.unregister_byte_stream_handler("lk.agent.pre-connect-audio-buffer")
+        # no pre-connect audio, so we generate a reply immediately
+        agent.generate_reply()
```

Comment:
this is the wrong example to apply this in. if the agent is going to speak first anyways then the instant audio doesn't really matter. for the purposes of this prototype i'd take ou the generate_reply entirely